### PR TITLE
Split up functor

### PR DESCRIPTION
**Pull request by adz**
Squashed and finished off discussed changes:
1) Add nowarn to avoid  operator warnings of `&&&` and `|||` for arrows.
2) Split out CE stuff and Alternative stuff out of Functor
3) Recombine InvariantFunctor into Functor

Hopefully in a better state now!

### CODE REVIEW

File: src/FSharpPlus/Control/ArrowApply.fs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#if !FABLE_COMPILER
+type App with
+    static member inline App (_output: '``ArrowApply<('ArrowApply<'T,'U> * 'T)>,'U)>``, _mthd: Default1) = App.InvokeOnInstance () : '``ArrowApply<('ArrowApply<'T,'U> * 'T)>,'U)>``
+    static member inline App (_output: ^t when ^t : null and ^t : struct              , _mthd: Default1) = id
+#endif
+
+
+namespace FSharpPlus.Internals
+module internal MonadOps =
```
Comment (adz) [inline src/FSharpPlus/Control/ArrowApply.fs]: As discussed in gitter: not sure of exact  implications.

FSharpPlus.Internals.MonadOps is opened by Indexable, MonadTrans and Traversable.

**Comment by adz** [comment]
Last things left to do:
1) Move out Set2 to Internal
2) Move out MonadOps

`module internal MonadOps =`

Right now I left it in place at bottom of ArrowApply.fs.

**Comment by gusty** [comment]
I would say if possible, move Applicative out of Monad.fs
An Applicative is not always a Monad, though the opposite is always true.

**Comment by adz** [comment]
@gusty moved out MonadOps to own file, but left intact. Also separated Applicative.

Seems you force pushed to master a little doc change, but was easy for me to reset to that then rebase...

Let me know if this looks ok, and I'll squash merge.

**Pull request merged**